src/Microsoft.Data.SqlClient/tests/ManualTests/SQL/TransactionTest/TransactionTest.cs

```diff
@@ -234,24 +234,7 @@ private void ExceptionTest()
                     SqlTransaction tx = connection.BeginTransaction();
 
                     string invalidSaveStateMessage = SystemDataResourceManager.Instance.SQL_NullEmptyTransactionName;
-                    string executeCommandWithoutTransactionMessage = SystemDataResourceManager.Instance.ADP_TransactionRequired("ExecuteNonQuery");
-                    string transactionConflictErrorMessage = SystemDataResourceManager.Instance.ADP_TransactionConnectionMismatch;
                     string parallelTransactionErrorMessage = SystemDataResourceManager.Instance.ADP_ParallelTransactionsNotSupported("SqlConnection");
-                    DataTestUtility.AssertThrowsWrapper<InvalidOperationException>(() =>
-                    {
-                        SqlCommand command = new SqlCommand("sql", connection);
-                        command.ExecuteNonQuery();
-                    }, executeCommandWithoutTransactionMessage);
-
-                    DataTestUtility.AssertThrowsWrapper<InvalidOperationException>(() =>
-                    {
-                        SqlConnection con1 = new SqlConnection(_connectionString);
-                        con1.Open();
-
-                        SqlCommand command = new SqlCommand("sql", con1);
-                        command.Transaction = tx;
-                        command.ExecuteNonQuery();
-                    }, transactionConflictErrorMessage);
```

Comment:
Please explain why this test was deleted?

Reply:
First remove test expected to fail for other reasons. Now since the transaction is handled internally this error `executeCommandWithoutTransactionMessage ` cannot occur.

Second test checks for bad assignments but expected to fail on `ExecuteNonQuery` while now it fails on `command.Transaction = tx`. So I removed the tests and added the equivalent and multiple variations in the new T3 prefixed tests I added for this change. Check those tests and you'll see this exact situation tested: assigning wrong connection's transaction, assigning closed transaction, and so on.

This is the path for the new tests:
`src\Microsoft.Data.SqlClient\tests\ManualTests\SQL\T3Test`

Reply:
Ok, please rename T3Test to something intuitive or move the tests to existing TransactionTest class.

Reply:
I'd rather not mix with existing test code. I renamed the class to `TransactionInferenceTest` and moved it to `ManualTests\SQL\TransactionTest\TransactionInferenceTest.cs`. Also the file mentions the related issue number for further reference.